libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.EditActions.schema

```diff
@@ -15,25 +15,30 @@
             "description": "Optional id for the dialog"
         },
         "disabled": {
-            "$role": "expression",
-            "type": "boolean",
+            "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/booleanExpression",
             "title": "Disabled",
             "description": "Optional condition which if true will disable this action.",
             "examples": [
                 "user.age > 3"
             ]
         },
         "changeType": {
-            "$role": "expression",
-            "type": "string",
+            "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/stringExpression",
             "title": "Type of change",
             "description": "Type of change to apply to the current actions.",
-            "enum": [
-                "insertActions",
-                "insertActionsBeforeTags",
-                "appendActions",
-                "endSequence",
-                "replaceSequence"
+            "oneOf": [
+                {
+                    "enum": [
+                        "insertActions",
+                        "insertActionsBeforeTags",
+                        "appendActions",
+                        "endSequence",
+                        "replaceSequence"
+                    ]
+                },
+                {
+                    "pattern": "^=(\\S){1}"
```

Comment:
>"pattern": "^=(\\S){1}" [](start = 20, length = 23)

This does not mean what you want.  Should be: ^=([^=]*)$

Reply:
I'm pretty sure it does, I tried it in multiple regex explorers and also verifiedin the schema validation that vscode does.  In fact, this was key to finding and discovering places in our .dialog files where it didn't follow the pattern.  What this regex says is that
a. start of the line MUST strt with an =
b. there must be at LEAST one non-whitespace char that follows.
Is there a reason you think it's wrong?  

---
In reply to: [410953123](https://github.com/microsoft/botbuilder-dotnet/pull/3758#discussion_r410953123) [](ancestors = 410953123)